### ISSUE TITLE
Commands Not Executing From amxx.cfg

### ISSUE THREAD

**Issue by Dr-Strange91** [alliedmodders/amxmodx#873]
# Environment
  * Operating System version:  Linux
  * Game/AppID (with version if applicable): CS 1.6
  * Current AMX Mod X version:  1.90.5270
  * Current Metamod version: Metamod-r 1.3.0.128

# Description

Since updating server from AMXX 1.90.529 to AMXX 1.90.5270, I can't execute following script (warmup.cfg) in comments from amxx.cfg, where as when same script if load (exec) from server.cfg works fine, which shows problem is with amxx.

Q: Then why don't I exec from server.cfg?
Ans: Server.cfg is only loaded on server restart, hence works only once where as amxx.cfg loads on every map change.

Q: Is Plugin Enabled?
Ans: Yes, Task Scheduler From Justin Homi is Loaded.

Q: Are all plugins of 1.90.5270?
Ans: Yes all except Admin Commands (admincmd.amxx)[1.90.5268]

Edit: Tried exec from amxx.

exec cstrike/warmup.cfg
exec cstrike/addons/amxmodx/configs/warmup.cfg

Both didn't work also, yes files existed at right places too.

# More Information

* https://dev-cs.ru/threads/11711/

# Script

```
// Warm UP Configuration

// show
amx_task 20 "amx_csay green PRE-WARM UP FOR: 5 seconds" s
amx_task 20 "amx_say PRE-WARM UP FOR: 5 seconds" s
amx_task 20 "amx_csay yellow PRE-WARM UP - NOT LIVE!" s
amx_task 21 "amx_say PRE-WARM UP - LIVE FOR: 5" s
amx_task 22 "amx_say PRE-WARM UP - LIVE FOR: 4" s
amx_task 23 "amx_say PRE-WARM UP - LIVE FOR: 3" s
amx_task 24 "amx_say PRE-WARM UP - LIVE FOR: 2" s
amx_task 25 "amx_say PRE-WARM UP - LIVE FOR: 1" s

amx_task 26 "sv_restart 1" s
amx_task 30 "amx_csay yellow WARMING - NOT LIVE !" s
amx_task 30 "amx_say WARMING - NOT LIVE !" s
amx_task 30 "amx_say WARMING - NOT LIVE !" s
amx_task 30 "amx_say WARMING - NOT LIVE !" s
amx_task 30 "amx_say WARMING - NOT LIVE !" s
amx_task 30 "amx_say WARMING - NOT LIVE !" s
amx_task 35 "amx_csay green LIVE FOR: 15" s
amx_task 35 "amx_say LIVE FOR: 15 seconds" s
amx_task 36 "amx_csay green LIVE FOR: 14" s
amx_task 37 "amx_csay green LIVE FOR: 13" s
amx_task 38 "amx_csay green LIVE FOR: 12" s
amx_task 39 "amx_csay green LIVE FOR: 11" s
amx_task 40 "amx_csay green LIVE FOR: 10" s
amx_task 40 "amx_say LIVE FOR: 10 seconds" s
amx_task 41 "amx_csay green LIVE FOR: 9" s
amx_task 42 "amx_csay green LIVE FOR: 8" s
amx_task 43 "amx_csay green LIVE FOR: 7" s
amx_task 44 "amx_csay green LIVE FOR: 6" s
amx_task 45 "amx_csay green LIVE FOR: 5" s
amx_task 45 "amx_say LIVE FOR: 5 seconds" s
amx_task 46 "amx_csay green LIVE FOR: 4" s
amx_task 47 "amx_csay green LIVE FOR: 3" s
amx_task 48 "amx_csay green LIVE FOR: 2" s
amx_task 49 "amx_csay green LIVE FOR: 1" s

amx_task 50 "sv_restart 1" s
amx_task 55 "amx_say LIVE LIVE LIVE !!! GL ^ HF by PSL GAMING COMMUNITY !!" s
amx_task 60 "amx_say LIVE LIVE LIVE !!! GL ^ HF !" s
amx_task 60 "amx_say LIVE LIVE LIVE !!! GL ^ HF !" s
amx_task 60 "amx_say LIVE LIVE LIVE !!! GL ^ HF !" s
amx_task 65 "amx_say For Adminship/Unban Requests, Visit: facebook.com/groups/PSLGaming/" s
```

**Comment by Arkshine** [comment]
Try the latest build, I've reverted [this code](https://github.com/alliedmodders/amxmodx/commit/b26118006a5b4f45350e72aa1a98f3372d1e08ad) in `admincmd.sma` for now.

**Comment by Dr-Strange91** [comment]
Hi, Thanks for helping that much quickly, Its almost 3 A.M here and in morning I have 24-32 hours job, hence will test it as soon as possible as back from work.

Sorry for responding late in regards to your quick help.

But anyone else in mean while can co-operate.

Well, the root of the problem was known and I hope current revert will fix it.

Thanks !!

**Comment by Dr-Strange91** [comment]
Hi, I updated the server to 1.90.5271, I followed script and tried to load it via 2 methods.

1. Directly putting it in amxx.cfg
2. Putting it in Configs/warmup.cfg & exec cstrike/addons/amxmodx/configs/warmup.cfg in amxx.cfg

Both ways failed to run the script.

I would request to review changes made since 1.90.5263.

Well as per my side I updated server from 1.90.5249 to 1.90.5270 (with admincmd (5268) & now 1.90.5271 (everything), yet this problem persists.

**Comment by Arkshine** [comment]
Actually, my comment was meant to be for #806 

About the issue here, I'm not sure. It's not clear and lacks information (you should give exact output of common commands such as `version`, `meta list`, `amxx version`, `amxx list`, `amxx modules`, etc. It's more reliable and helpful).

Are you saying `amxx.cfg` is not called? 
There is an echo inside (`Executing AMX Mod X Configuration File`), you should be able to see in the _server_ console/log.

I did try myself with the latest AMXX version under windows and with your plugin, and the commands inside `amxx.cfg` are well called.

**Comment by Dr-Strange91** [comment]
Hi, @Arkshine Welcome Back !!

As per my 2nd reply, I clarified amxx.cfg do called but the warmup script dosen't load, I tried exec it executed but didn't work, I also tried direct putting script into amxx.cfg too that also failed to work.

Its like cvars are not being called.

```
AMX Mod X 1.9.0.5271 (http://www.amxmodx.org)
Authors:
	David "BAILOPAN" Anderson, Pavol "PM OnoTo" Marko
	Felix "SniperBeamer" Geyer, Jonny "Got His Gun" Bergstrom
	Lukasz "SidLuke" Wlasinski, Christian "Basic-Master" Hammacher
	Borja "faluco" Ferrer, Scott "DS" Ehlert
Compiled: Jun 24 2020 13:53:30
Built from: https://github.com/alliedmodders/amxmodx/commit/5eea3e5
Build ID: 5271:5eea3e5
Core mode: JIT+ASM32

```
```
Metamod-r v1.3.0.128, API (5:13)
Metamod-r build: 17:47:54 Aug 24 2018
Metamod-r from: https://github.com/theAsmodai/metamod-r/commit/0cf2f70
```
```
Protocol version 48
Exe version 1.1.2.7/Stdio (cstrike)
ReHLDS version: 3.7.0.696-dev
Build date: 18:46:47 May 18 2020 (2228)
Build from: https://github.com/dreamstalker/rehlds/commit/987ee51
```
```
ReGameDLL version: 5.18.0.470-dev
Build date: 16:42:40 Jun 17 2020
Build from: https://github.com/s1lentq/ReGameDLL_CS/commit/3878f46
```
```
Currently loaded plugins:
       name                    version     author            file             status   
 [  1] My Ban System             1.1.2       K@[L]@uz          kalauz_ban_syst  running  
 [  2] Admin Commands Log      1.2.1       twisterniq        admin_commands_  running  
 [  3] ServerNamePlay          1.8.2       Rock Rocky        PSL_hostname.am  running  
 [  4] Scan_Say                12/May/202  General Memer     scan_say.amxx    running  
 [  5] Task Scheduler          0.2         JustinHoMi        task_scheduler.  running  
 [  6] Mode 2x2                2.5re-19-0  s1lent            PSL_mode_edit.a  running  
 [  7] Online Logger           06.04.18    mx?!              PSL_online_logg  running  
 [  8] Free Look Crosshair Fi  1.3         Numb              free_look_cross  running  
 [  9] Admin Base              1.9.0.5271  AMXX Dev Team     admin.amxx       running  
 [ 10] Admin Commands          1.9.0.5271  AMXX Dev Team     admincmd.amxx    running  
 [ 11] Admin Help              1.9.0.5271  AMXX Dev Team     adminhelp.amxx   running  
 [ 12] Multi-Lingual System    1.9.0.5271  AMXX Dev Team     multilingual.am  running  
 [ 13] Menus Front-End         1.9.0.5271  AMXX Dev Team     menufront.amxx   running  
 [ 14] Commands Menu           1.9.0.5271  AMXX Dev Team     cmdmenu.amxx     running  
 [ 15] New KickMenu            2.0         Flash             kickmenu.amxx    running  
 [ 16] Players Menu            1.9.0.5271  AMXX Dev Team     plmenu.amxx      running  
 [ 17] Maps Menu               1.9.0.5271  AMXX Dev Team     mapsmenu.amxx    running  
 [ 18] Plugin Menu             1.9.0.5271  AMXX Dev Team     pluginmenu.amxx  running  
 [ 19] Chat Manager            1.1.1-11    Mistrick          PSL_GREEN_chatm  running  
 [ 20] Chat Manager: Addon     0.0.4-70    Mistrick          PSL_chatmanager  running  
 [ 21] Anti Flood              1.9.0.5271  AMXX Dev Team     antiflood.amxx   running  
 [ 22] Admin Votes             1.9.0.5271  AMXX Dev Team     adminvote.amxx   running  
 [ 23] NextMap                 1.9.0.5271  AMXX Dev Team     nextmap.amxx     running  
 [ 24] Roundsleft              0.2         JGHG              roundsleft_nosp  running  
 [ 25] Stats Configuration     1.9.0.5271  AMXX Dev Team     statscfg.amxx    running  
 [ 26] Restrict Weapons        1.9.0.5271  AMXX Dev Team     restmenu.amxx    running  
 [ 27] StatsX                  1.9.0.5271  AMXX Dev Team     statsx.amxx      running  
 [ 28] CS Misc. Stats          1.9.0.5270  AMXX Dev Team     miscstats.amxx   running  
 [ 29] CS Stats Logging        1.9.0.5271  AMXX Dev Team     stats_logging.a  running  
 [ 30] Accuracy Fix            3.0         Numb              accuracy_fix.am  running  
 [ 31] [PSL] Admin Color Chat  1.0         Edit By Alber     PSL_admin_color  running  
 [ 32] Admin Chat              1.9.0.5271  AMXX Dev Team     adminchat.amxx   running  
 [ 33] Team Balancer           1.8b3       Ptahhotep         ptb.amxx         running  
 [ 34] Block IP Messages       1.0 Alpha   Alik Inline Asla  PSL_BlockIpInNi  running  
 [ 35] Admin Spectator ESP     1.5         KoST              admin_spectator  running  
 [ 36] FragCounter             1.01        Scarzzurs         fragcounter.amx  running  
 [ 37] KGB Bots                2.3         OvidiuS & Desika  kgbbots.amxx     running  
 [ 38] Steamers Info           1.0         K@[L]@uz          steamers_info.a  running  
 [ 39] Map Spawns Editor       1.0.16      iG_os             Map_Spawns_Edit  running  
 [ 40] Reset Score             0.0.2f      wopox1337         PSL_ResetScore.  running  
 [ 41] Fast Sniper Switch      1.3         Numb              fast_sniper_swi  running  
 [ 42] Spec Money Save         0.2         F@nt0M            spec_money_save  running  
 [ 43] AMX Slay Losers         1.1         d3n14@yahoo.com   slaylosers.amxx  running  
 [ 44] Autoresponder/Advertis  0.5         MaximusBrood      ad_manager5.amx  running  
 [ 45] Auto Join on Connect    0.1         VEN               auto_join_on_co  running  
 [ 46] Ping Faker              1.5a        MeRcyLeZZ         pingfaker.amxx   running  
 [ 47] c4 timer                1.1         cheap_suit        c4timer.amxx     running  
 [ 48] [PSL] Info Rounds By A  1.0         K@[L]@uz          PSL_inforounds.  running  
 [ 49] Descriptive 'Fire in t  0.1         VEN               colored_radio.a  running  
 [ 50] WalkGuard               1.3.2       mogel             walkguard.amxx   running  
 [ 51] VIP Eng Version         3.0         Dunno             PSL_VIP_FREE_WO  running  
 [ 52] AMX Client Exec         0.03        v3x               amx_exec.amxx    running  
 [ 53] CS AFK Manager          1.0.6 (amx  Freeman           afk_manager.amx  running  
 [ 54] AMXX Gag                1.5.0       xPaw & Exolent    amx_gag.amxx     running  
 [ 55] Boost Server            1.0         K@[L]@UZ          boost_server.am  running  
 [ 56] Invisible Spectator     0.2         ConnorMcLeod      PSL_Invis_Spec.  running  
 [ 57] unknown                 unknown     unknown           pslgamename.amx  running  
 [ 58] [ReAPI] No Team Flash   0.0.3       Vaqtincha         no_team_flash.a  running  
 [ 59] ReAimDetector API       0.2.1       ReHLDS Team       reaimdetector.a  running  
 [ 60] Permanent message       1.1         He3aBuCuM         permanent_messa  running  
 [ 61] Played Time             1.4         Alka > Edit By A  PSL_played_time  running  
 [ 62] Anti sv_password Chang  1.0         BesT              autopassremover  running  
 [ 63] Block Advertising       1.3         Alka              block_advertisi  running  
 [ 64] Block IP                1.0         bad_boy           Anti_IP_With_Sp  running  
 [ 65] xREDIRECT               1.0.3.3     x0R               xredirect.amxx   running  
65 plugins, 65 running
```
```
Currently loaded plugins:
      description       stat pend  file                        vers             src  load  unload
 [ 1] SafeNameAndChat   RUN   -    SafeNameAndChat.so          v1.1             ini  ANY   ANY  
 [ 2] Reunion           RUN   -    reunion_mm_i386.so          v0.1.0.135       ini  Start Never
 [ 3] AMX Mod X         RUN   -    amxmodx_mm_i386.so          v1.9.0.5270      ini  Start ANY  
 [ 4] Print Center Fix  RUN   -    printcenterfix_mm_i386.so   v1.0.1           ini  ANY   Never
 [ 5] Revoice           RUN   -    revoice_mm_i386.so          v0.1.0.34        ini  Start Never
 [ 6] ReSemiclip        RUN   -    resemiclip_mm_i386.so       v2.3.9           ini  Chlvl ANY  
 [ 7] WHBlocker         RUN   -    whblocker_mm_i386.so        v1.5.697         ini  Chlvl ANY  
 [ 8] Fun               RUN   -    fun_amxx_i386.so            v1.9.0.5270      pl3  ANY   ANY  
 [ 9] Engine            RUN   -    engine_amxx_i386.so         v1.9.0.5270      pl3  ANY   ANY  
 [10] FakeMeta          RUN   -    fakemeta_amxx_i386.so       v1.9.0.5270      pl3  ANY   ANY  
 [11] GeoIP             RUN   -    geoip_amxx_i386.so          v1.9.0.5270      pl3  ANY   ANY  
 [12] CStrike           RUN   -    cstrike_amxx_i386.so        v1.9.0.5270      pl3  ANY   ANY  
 [13] CSX               RUN   -    csx_amxx_i386.so            v1.9.0.5270      pl3  ANY   ANY  
 [14] Ham Sandwich      RUN   -    hamsandwich_amxx_i386.so    v1.9.0.5270      pl3  ANY   ANY  
 [15] ReAPI             RUN   -    reapi_amxx_i386.so          v5.14.0.195-dev  pl3  ANY   Never
 [16] hackdetector      RUN   -    hackdetector_amxx_i386.so   v0.15.328.lite   pl3  ANY   ANY  
 [17] ReAimDetector     RUN   -    reaimdetector_amxx_i386.so  v0.2.2           pl3  ANY   Never
 [18] MySQL             RUN   -    mysql_amxx_i386.so          v1.9.0.5270      pl3  ANY   ANY  
18 plugins, 18 running
```

**Comment by Arkshine** [comment]
It would have been easier to understand if you said that your issue is only with `exec`. I don't think the issue is related to AMXX itself.

Usually path is relative to the `addons`, so this one should work: `exec addons/amxmodx/configs/warmup.cfg`. 
It works for me under windows. Please give a try.

**Comment by Dr-Strange91** [comment]
Hi, its not working.

# Images
![Screenshot_2](https://user-images.githubusercontent.com/35621407/85924672-39fdaa80-b8ad-11ea-9aa0-5db337959dfd.png)
![Screenshot_1](https://user-images.githubusercontent.com/35621407/85924676-3e29c800-b8ad-11ea-8a8a-67f37e66a951.png)

# Video
[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/jxq-dpxNFDU/0.jpg)](https://www.youtube.com/watch?v=jxq-dpxNFDU)
P.S: Round is restarted by admin.

**Comment by Arkshine** [comment]
I did tested with `amxx.cfg`. 

Well, you can check yourself. In the server console(/log, if the file is not found, you will see "couldn't exec..." is the case here?
You can also put some echo in your file to make sure it's called.

**Comment by Dr-Strange91** [comment]
Thanks !! Bug is found, I tried checking all latest changes which included, Admin Command Logs (recent version) by @twisterniq and I found it was blocking Task Schedular by Justin Homi.

1st test: Removed Admin Commands Log > Issue Fixed.
2nd test: Added Commands Log >Issue Returned.
3rd test: Put Task Schedular above Admin Commands Log > Issue Fixed.

These all sorted. There was no problem with AMXX but plugin I had.

Sorry for wasting alot for your time & at the same time a bunch of Thanks & Respect for helping out till end.
@Arkshine Once Again Thanks alot.

With it, I will close the issue.

P.S: Regarding your recent reply, Yes I was trying echo warmup.cfg is being executed.
